clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1005,8 +1003,22 @@ public void assign(Collection<TopicPartition> partitions) {
         // send any new fetches (won't resend pending fetches)
         fetcher.sendFetches();
 
+        // if no fetches could be sent at the moment (which can happen if a partition leader is in the
+        // blackout period following a disconnect), then we only block for the retry backoff duration.
+        if (!fetcher.hasInFlightFetches())
+            timeout = retryBackoffMs;
```

Comment:
Ack on min(). I doubt using the backoff matters and trying to get nearer the mark would add some additional complexity. We could also hit this case if a partition leader is unknown (or, looking at the code, if we happen to try to fetch from a broker which already has a metadata request in progress), so I'd rather keep the backoff logic a little stupid to be sure that we've hit all the cases.